ecl/hql/hqlgram2.cpp

```diff
@@ -6171,8 +6171,11 @@ void HqlGram::reportErrorUnexpectedX(const attribute& errpos, IAtom * unexpected
 
 void HqlGram::doReportWarning(WarnErrorCategory category, int warnNo, const char *msg, const char *filename, int lineno, int column, int pos)
 {
-    Owned<IError> error = createError(category, queryDefaultSeverity(category), warnNo, msg, filename, lineno, column, pos);
-    report(error);
+	if (!lookupCtx.syntaxCheckingUsingCache() || category==CategorySyntax || category==CategoryError)
```

Comment:
minor: Consider including in a function since this code is repeated 5 times.  Adding a function can also aid readability.  E.g. if (includeError(lookupCtx, category))